### PR TITLE
Adding SQL Batch operation & Parameterised Query BBEs

### PR DESCRIPTION
**Pull request by niveathika**
## Purpose
$subject

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: examples/jdbc2-batch-execute-operation/jdbc2_batch_execute_operation.bal [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        "country VARCHAR(300), PRIMARY KEY (customerId))");
+    if (result is sql:ExecutionResult) {
+        io:println("Create table executed: ", result);
+    }
+
+}
+
+function insertRecords(jdbc:Client jdbcClient) returns int[] {
+
+    // Records to be inserted
```
Comment (praneesha) [inline examples/jdbc2-batch-execute-operation/jdbc2_batch_execute_operation.bal:29]:
```suggestion
    // Records to be inserted.
```

File: examples/jdbc2-batch-execute-operation/jdbc2_batch_execute_operation.bal [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        io:println("Error occured: ", result);
+    } else {
+        io:println("Empty result");
+    }
+    return generatedIds;
+}
+
+function updateRecords(jdbc:Client jdbcClient, int[] generatedIds) {
+
+    // Update Queries
```
Comment (praneesha) [inline examples/jdbc2-batch-execute-operation/jdbc2_batch_execute_operation.bal:70]:
```suggestion
    // The update queries.
```

File: examples/jdbc2-batch-execute-operation/jdbc2_batch_execute_operation.bal [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    if (jdbcClient is jdbc:Client) {
+        // Initialize a table and insert sample data.
+        sql:Error? initResult = initializeTable(jdbcClient);
+        if (initResult is ()) {
+            // Insert a batch of records.
+            int[] generatedIds = insertRecords(jdbcClient);
+            // Update records.
+            updateRecords(jdbcClient, generatedIds);
+
+            // Check data.
```
Comment (praneesha) [inline examples/jdbc2-batch-execute-operation/jdbc2_batch_execute_operation.bal:106]:
```suggestion
            // Check the data.
```

File: examples/mysql-batch-execute-operation/mysql_batch_execute_operation.bal [hunk 4]
```diff
@@ -0,0 +1,158 @@
+import ballerina/io;
+import ballerina/mysql;
+import ballerina/sql;
+
+// Username and password of the MySQL database. This is used in the below
```
Comment (praneesha) [inline examples/mysql-batch-execute-operation/mysql_batch_execute_operation.bal:5]:
```suggestion
// The username and password of the MySQL database. This is used in the below
```

File: examples/mysql-batch-execute-operation/mysql_batch_execute_operation.bal [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+// examples when initializing the MySQL connector. You need to change these
+// based on your setup if you try locally.
+string dbUser = "root";
+string dbPassword = "Test@123";
+string dbName = "MYSQL_BBE_EXEC";
+
+function initializeDatabase() returns sql:Error? {
+    // Initialize the client without any database to create the database.
+    mysql:Client mysqlClient = check new (user = dbUser, password = dbPassword);
+    // Create database if it does not exist. If any error occurred,
```
Comment (praneesha) [inline examples/mysql-batch-execute-operation/mysql_batch_execute_operation.bal:15]:
```suggestion
    // Create the database if it does not exist. If any error occurred,
```

File: examples/mysql-batch-execute-operation/mysql_batch_execute_operation.bal [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+        "country VARCHAR(300), PRIMARY KEY (customerId))");
+    if (result is sql:ExecutionResult) {
+        io:println("Create table executed: ", result);
+    }
+
+}
+
+function insertRecords(mysql:Client mysqlClient) returns int[] {
+
+    // Records to be inserted
```
Comment (praneesha) [inline examples/mysql-batch-execute-operation/mysql_batch_execute_operation.bal:48]:
```suggestion
    // Records to be inserted.
```

File: examples/mysql-batch-execute-operation/mysql_batch_execute_operation.bal [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+        io:println("Error occured: ", result);
+    } else {
+        io:println("Empty result");
+    }
+    return generatedIds;
+}
+
+function updateRecords(mysql:Client mysqlClient, int[] generatedIds) {
+
+    // Update Queries
```
Comment (praneesha) [inline examples/mysql-batch-execute-operation/mysql_batch_execute_operation.bal:89]:
```suggestion
    // The update queries.
```

File: examples/mysql-batch-execute-operation/mysql_batch_execute_operation.bal [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+        if (mysqlClient is mysql:Client) {
+            // Initialize a table and insert sample data.
+            sql:Error? initResult = initializeTable(mysqlClient);
+            if (initResult is ()) {
+                // Insert a batch of records.
+                int[] generatedIds = insertRecords(mysqlClient);
+                // Update records.
+                updateRecords(mysqlClient, generatedIds);
+
+                // Check data.
```
Comment (praneesha) [inline examples/mysql-batch-execute-operation/mysql_batch_execute_operation.bal:129]:
```suggestion
                // Check the data.
```

File: examples/jdbc-batch-execute-operation/jdbc_batch_execute_operation.metatags [hunk 9]
```diff
@@ -0,0 +1,2 @@
+description: BBE on the use the JDBC client to execute a batch of DDL/DML operations.
```
Comment (Samallama) [inline examples/jdbc-batch-execute-operation/jdbc_batch_execute_operation.metatags:1]:
```suggestion
description: BBE on how to use the JDBC client to execute a batch of DDL/DML operations in Ballerina.
```

File: examples/jdbc-complex-type-queries/jdbc_complex_type_queries.metatags [hunk 10]
```diff
@@ -0,0 +1,2 @@
+description: BBE on the use of JDBC client with complex data types such as arrays, binary, blobs, clob, and date/time fields.
```
Comment (Samallama) [inline examples/jdbc-complex-type-queries/jdbc_complex_type_queries.metatags:1]:
```suggestion
description: BBE on the use of JDBC client with complex data types such as arrays, binary, blobs, clob, and date/time fields in Ballerina.
```

File: examples/jdbc-execute-operation/jdbc_execute_operation.metatags [hunk 11]
```diff
@@ -0,0 +1,2 @@
+description: BBE on the use of JDBC client with DDL and DML operations.
```
Comment (Samallama) [inline examples/jdbc-execute-operation/jdbc_execute_operation.metatags:1]:
```suggestion
description: BBE on the use of JDBC client with DDL and DML operations in Ballerina.
```

File: examples/jdbc-init-options/jdbc_init_options.metatags [hunk 12]
```diff
@@ -0,0 +1,2 @@
+description: BBE demonstrates how to initialize the JDBC client with different initialization options.
```
Comment (Samallama) [inline examples/jdbc-init-options/jdbc_init_options.metatags:1]:
```suggestion
description: BBE demonstrates how to initialize the JDBC client with different initialization options in Ballerina.
```

File: examples/jdbc-parameterized-query/jdbc_parameterized_query.description [hunk 13]
```diff
@@ -0,0 +1,6 @@
+// This example demonstrates how to use `sql:ParameterizedQuery` to easily
+// create pre compiled SQL statements which can be reused. Note that the
```
Comment (Samallama) [inline examples/jdbc-parameterized-query/jdbc_parameterized_query.description:2]:
```suggestion
// create pre-compiled SQL statements that can be reused. Note that the 
```

File: examples/jdbc-query-operation/jdbc_query_operation.metatags [hunk 14]
```diff
@@ -0,0 +1,2 @@
+description: BBE demonstrates how to use the JDBC client select query operations with the stream return type.
```
Comment (Samallama) [inline examples/jdbc-query-operation/jdbc_query_operation.metatags:1]:
```suggestion
description: BBE demonstrates how to use the JDBC client select query operations with the stream return type in Ballerina.
```

File: examples/mysql-batch-execute-operation/mysql_batch_execute_operation.metatags [hunk 15]
```diff
@@ -0,0 +1,2 @@
+description: BBE on the use the MySQL client to execute a batch of DDL/DML operations.
```
Comment (Samallama) [inline examples/mysql-batch-execute-operation/mysql_batch_execute_operation.metatags:1]:
```suggestion
description: BBE on how to use the MySQL client to execute a batch of DDL/DML operations in Ballerina.
```

File: examples/mysql-complex-type-queries/mysql_complex_type_queries.metatags [hunk 16]
```diff
@@ -0,0 +1,2 @@
+description: BBE on the use of MySQL client with complex data types such as arrays, binary, blobs, clob, and date/time fields.
```
Comment (Samallama) [inline examples/mysql-complex-type-queries/mysql_complex_type_queries.metatags:1]:
```suggestion
description: BBE on the use of MySQL client with complex data types such as arrays, binary, blobs, clob, and date/time fields in Ballerina.
```

File: examples/mysql-execute-operation/mysql_execute_operation.metatags [hunk 17]
```diff
@@ -0,0 +1,2 @@
+description: BBE on the use of MySQL client with DDL and DML operations.
```
Comment (Samallama) [inline examples/mysql-execute-operation/mysql_execute_operation.metatags:1]:
```suggestion
description: BBE on the use of MySQL client with DDL and DML operations in Ballerina.
```

File: examples/mysql-init-options/mysql_init_options.metatags [hunk 18]
```diff
@@ -0,0 +1,2 @@
+description: BBE demonstrates how to initialize the MySQL client with different initialization options.
```
Comment (Samallama) [inline examples/mysql-init-options/mysql_init_options.metatags:1]:
```suggestion
description: BBE demonstrates how to initialize the MySQL client with different initialization options in Ballerina.
```

File: examples/mysql-parameterized-query/mysql_parameterized_query.description [hunk 19]
```diff
@@ -0,0 +1,6 @@
+// This example demonstrates how to use `sql:ParameterizedQuery` to easily
+// create pre compiled SQL statements which can be reused. Note that the
```
Comment (Samallama) [inline examples/mysql-parameterized-query/mysql_parameterized_query.description:2]:
```suggestion
// create pre-compiled SQL statements that can be reused. Note that the 
```

File: examples/mysql-parameterized-query/mysql_parameterized_query.metatags [hunk 20]
```diff
@@ -0,0 +1,2 @@
+description: BBE demonstrates how to use `sql:ParameterizedQuery` to easily create pre compiled SQL statements which can be reused.
```
Comment (Samallama) [inline examples/mysql-parameterized-query/mysql_parameterized_query.metatags:1]:
```suggestion
description: BBE demonstrates how to use `sql:ParameterizedQuery` in Ballerina to easily create pre-compiled SQL statements that can be reused.
```

File: examples/mysql-parameterized-query/mysql_parameterized_query.metatags [hunk 21]
```diff
@@ -0,0 +1,2 @@
...
+keywords: ballerina, ballerina by example, bbe, mysql, parameterized query, parameters, update, insert, delete
```
Comment (Samallama) [inline examples/mysql-parameterized-query/mysql_parameterized_query.metatags:2]: Shall we remove one keyword? The best practice is to have a maximum of 5-8 keywords.

File: examples/mysql-query-operation/mysql_query_operation.metatags [hunk 22]
```diff
@@ -0,0 +1,2 @@
+description: BBE demonstrates how to use the MySQL client select query operations with the stream return type.
```
Comment (Samallama) [inline examples/mysql-query-operation/mysql_query_operation.metatags:1]:
```suggestion
description: BBE demonstrates how to use the MySQL client select query operations with the stream return type in Ballerina.
```

File: examples/mysql-parameterized-query/mysql_parameterized_query.metatags (hunk 21 above)
Comment (Samallama) [reply examples/mysql-parameterized-query/mysql_parameterized_query.metatags:2]: Please see above comment

Comment (niveathika) [reply examples/mysql-parameterized-query/mysql_parameterized_query.metatags]: Fixed in ab7ae07

File: examples/jdbc-parameterized-query/jdbc_parameterized_query.description [hunk 23]
```diff
@@ -0,0 +1,8 @@
+// This example demonstrates how to use `sql:ParameterizedQuery` to easily
```
Comment (praneesha) [inline examples/jdbc-parameterized-query/jdbc_parameterized_query.description:1]:
```suggestion
// This example demonstrates how to use the `sql:ParameterizedQuery` to easily
```

File: examples/mysql-parameterized-query/mysql_parameterized_query.description [hunk 24]
```diff
@@ -0,0 +1,8 @@
+// This example demonstrates how to use `sql:ParameterizedQuery` to easily
```
Comment (praneesha) [inline examples/mysql-parameterized-query/mysql_parameterized_query.description:1]:
```suggestion
// This example demonstrates how to use the `sql:ParameterizedQuery` to easily
```

File: examples/mysql-parameterized-query/mysql_parameterized_query.metatags [hunk 25]
```diff
@@ -0,0 +1,2 @@
+description: BBE demonstrates how to use `sql:ParameterizedQuery` in Ballerina to easily create pre-compiled SQL statements that can be reused.
```
Comment (praneesha) [inline examples/mysql-parameterized-query/mysql_parameterized_query.metatags:1]:
```suggestion
description: BBE demonstrates how to use the `sql:ParameterizedQuery` in Ballerina to easily create pre-compiled SQL statements that can be reused.
```

**Pull request merged**